src/components/Questionnaire/index.tsx

```diff
@@ -46,59 +74,162 @@ export function QuestionnaireList() {
         </Button>
       </div>
 
-      <div className="overflow-hidden rounded-lg bg-white shadow">
-        <table className="min-w-full divide-y divide-gray-200">
-          <thead className="bg-gray-50">
-            <tr>
-              <th className="px-6 py-3 text-left text-xs font-medium uppercase tracking-wider text-gray-500">
-                {t("title")}
-              </th>
-              <th className="px-6 py-3 text-left text-xs font-medium uppercase tracking-wider text-gray-500">
-                {t("description")}
-              </th>
-              <th className="px-6 py-3 text-left text-xs font-medium uppercase tracking-wider text-gray-500">
-                {t("status")}
-              </th>
-              <th className="px-6 py-3 text-left text-xs font-medium uppercase tracking-wider text-gray-500">
-                {t("slug")}
-              </th>
-            </tr>
-          </thead>
-          <tbody className="divide-y divide-gray-200 bg-white">
-            {questionnaireList.map((questionnaire: QuestionnaireDetail) => (
-              <tr
-                key={questionnaire.id}
-                onClick={() => navigate(`/questionnaire/${questionnaire.slug}`)}
-                className="cursor-pointer hover:bg-gray-50"
+      <div className="flex flex-wrap items-center my-2 gap-2">
+        <Popover>
+          <PopoverTrigger asChild>
+            <Button
+              variant="outline"
+              size="sm"
+              className={cn(
+                "h-8 min-w-[120px] justify-start",
+                title && "bg-primary/10 text-primary hover:bg-primary/20",
+              )}
+            >
+              <CareIcon icon="l-search" className="mr-2 h-4 w-4" />
+              {title ? <span className="truncate">{title}</span> : t("search")}
+            </Button>
+          </PopoverTrigger>
+          <PopoverContent
+            className="w-[20rem] p-3"
+            align="start"
+            onEscapeKeyDown={(event) => event.preventDefault()}
+          >
+            <div className="space-y-4">
+              <h4 className="font-medium leading-none">
+                {t("search_questionnaire")}
+              </h4>
+              <Input
+                id="questionnaire-search"
+                type="text"
+                placeholder={t("search_by_questionnaire_title")}
+                value={title}
+                onChange={(event) =>
+                  updateQuery({
+                    status,
+                    title: event.target.value,
+                  })
+                }
+              />
+            </div>
+          </PopoverContent>
+        </Popover>
+        <div className="items-center">
+          <Tabs value={status || "all"} className="w-full">
+            <TabsList className="bg-transparent p-0 h-8">
+              <TabsTrigger
```

Comment:
We can use an array of values and map those to a TabsTrigger for this, since code is the same.